colobot-base/src/ui/controls/window.cpp

```diff
@@ -154,15 +154,22 @@ CImage* CWindow::CreateImage(const glm::vec2& pos, const glm::vec2& dim, int ico
 
 // Creates a new label.
 
-CLabel* CWindow::CreateLabel(const glm::vec2& pos, const glm::vec2& dim, int icon, EventType eventMsg, std::string name)
+CLabel* CWindow::CreateLabel(const glm::vec2& pos, const glm::vec2& dim, int icon, EventType eventMsg, std::string name, bool removeTooltip)
 {
     CLabel* label = CreateControl<CLabel>(pos, dim, icon, eventMsg);
 
-    auto p = name.find("\\");
-    if (p == std::string::npos)
-        label->SetName(name);
+    if (removeTooltip)
+    {
+        auto p = name.find("\\");
```

Comment:
Do we actually rely on the tooltip-removing ability of this method? Where? How? Why? If this only displays the "main" part of the text, what displays the tooltip?

If we don't need this, we can just remove the whole find&substr part.

Reply:
From my understanding, labels remove the tooltip part of the text by default. This might be to allow sharing of texts between labels and other controls that do display tooltips. So if you had a button with text `"New player\\Create new player profile"`, a button would have `"New player"` text and `"Create new player profile"` tooltip. But if you set this to a label, it would only display `"New player"` and had no tooltip.

I don't know why it's done like this, I find this very clunky and confusing; tooltip should always be separate from the main text. But I wrote this code to preserve existing behavior, even if it's weird, in order to hopefully not introduce any new bugs.

Reply:
I found an example, where we do depend on tooltip removal

This appears to contain both the tooltip for the slider and the label:

https://github.com/colobot/colobot/blob/03b2da88e5b83016b3a15453d2cd10214eb7d317/colobot-base/src/common/restext.cpp#L243

https://github.com/colobot/colobot/blob/03b2da88e5b83016b3a15453d2cd10214eb7d317/colobot-base/src/ui/screen/screen_setup_game.cpp#L104-L110

![image](https://github.com/colobot/colobot/assets/52621858/f086cb20-2a33-4e8b-af75-0ffb038cc591)

Reply:
Turns out that refactoring SetName() will take a lot more work than I anticipated. I'd leave it the way it is, it can be addressed in a separate issue. I did the other change though.